src/wallet/wallet.cpp

```diff
@@ -966,8 +966,9 @@ bool CWallet::LoadToWallet(const CWalletTx& wtxIn)
  * Abandoned state should probably be more carefully tracked via different
  * posInBlock signals or by checking mempool presence when necessary.
  */
-bool CWallet::AddToWalletIfInvolvingMe(const CTransaction& tx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate)
+bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate)
```

Comment:
Now that `AddToWalletIfInvolvingMe()` is called with `pindexBlockConnected` set to NULL when the transaction isn't in a block, can we remove the magic -1 value for `posInBlock` (and change the `if (posInBlock != -1)` into `if (pindexBlockConnected != NULL)`)?